reactor-netty-http/src/test/java/reactor/netty/http/HttpMetricsHandlerTests.java

```diff
@@ -233,6 +233,59 @@ else if (clientProtocols.length == 2 &&
 				numWrites[1], bytesWrite[1]);
 	}
 
+	// https://github.com/reactor/reactor-netty/issues/2187
+	@ParameterizedTest
+	@MethodSource("httpCompatibleProtocols")
+	void testRecordingFailsServerSide(HttpProtocol[] serverProtocols, HttpProtocol[] clientProtocols,
+	                                  @Nullable ProtocolSslContextSpec serverCtx, @Nullable ProtocolSslContextSpec clientCtx) throws Exception {
+		disposableServer = customizeServerOptions(httpServer, serverCtx, serverProtocols)
+				.metrics(true, id -> {
+					throw new IllegalArgumentException("Testcase injected Exception");
+				})
+				.bindNow();
+
+		AtomicReference<SocketAddress> serverAddress = new AtomicReference<>();
+		httpClient = customizeClientOptions(httpClient, clientCtx, clientProtocols).doAfterRequest((req, conn) ->
+            serverAddress.set(conn.channel().remoteAddress())
+		);
```

Comment:
`serverAddress` is never used

```suggestion
		httpClient = customizeClientOptions(httpClient, clientCtx, clientProtocols);
```

Reply:
Removed unused variable